File: algorithms/python3/replaceElementsWithGreatestElementOnRightSide.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        temp = arr[len(arr) - 1]
        arr[len(arr) - 1] = -1
        for i in range(len(arr) - 2, -1, -1):
            current = arr[i]
            arr[i] = temp
            if (current > temp):
                temp = current
        return arr

File: algorithms/c/replaceElementsWithGreatestElementOnRightSide.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* replaceElements(int* arr, int arrSize, int* returnSize){
    *returnSize = arrSize;
    int temp = arr[arrSize - 1];
    arr[arrSize - 1] = -1;
    for (int i = arrSize - 2; i >= 0; i--) {
        int current = arr[i];
        arr[i] = temp;
        if (current > temp) {
            temp = current;
        }
    }
    return arr;
}
